src/main/java/com/salesforce/storm/spout/dynamic/metrics/LogRecorder.java

```diff
@@ -42,36 +39,13 @@
 public class LogRecorder implements MetricsRecorder {
 
     private static final Logger logger = LoggerFactory.getLogger(LogRecorder.class);
-    private final Map<String, Long> counters = new ConcurrentHashMap<>();
-    private final Map<String, Object> assignedValues = new ConcurrentHashMap<>();
-
-    // For storing timer start values
-    private final Map<String, Long> timerStartValues = new ConcurrentHashMap<>();
-
-    @Override
-    public void open(Map spoutConfig, TopologyContext topologyContext) {
-    }
 
-    @Override
-    public void close() {
-        // Noop
-    }
+    private final KeyBuilder keyBuilder = new KeyBuilder(null);
```

Comment:
Should this be set in open()?  Is KeyBuilder serializable?

Reply:
Hmmm... good call. It probably serializes fine, but this is a good reason to put it in open(), though I hope no one uses this recorder in production.